File: python/2859.py
from typing import List

class Solution:
    def sumIndicesWithKSetBits(self, nums: List[int], k: int) -> int:
        def bitCount(num: int) -> int:
            cnt = 0
            while num:
                cnt+=1
                num -= num &~ (num-1)
            return cnt
        res = 0
        for i, num in enumerate(nums):
            if k == bitCount(i):
                res += num
        return res

File: go/2859.go
package main

func sumIndicesWithKSetBits(nums []int, k int) (res int) {
	var bitCount func(int) int
	bitCount = func(num int) (cnt int) {
		for num > 0 {
			cnt++
			num -= num & ^(num - 1)
		}
		return
	}
	for i, num := range nums {
		if bitCount(i) == k {
			res += num
		}
	}
	return
}
